src/stravalib/client.py

```diff
@@ -783,22 +797,29 @@ def create_activity(
             elapsed_time=elapsed_time,
         )
 
-        if description is not None:
-            params["description"] = description
+        update_params = {
+            "description": description,
+            "distance": distance,
+            "commute": commute,
+            "trainer": trainer,
+        }
 
-        if distance is not None:
-            params["distance"] = distance
+        for key, value in update_params.items():
+            if value is not None:
+                params[key] = value
 
         params = self._validate_activity_type(
             params, activity_type, sport_type
         )
 
         raw_activity = self.protocol.post("/activities", **params)
 
-        return model.Activity.model_validate(
+        return model.DetailedActivity.model_validate(
             {**raw_activity, **{"bound_client": self}}
         )
 
+    # TODO: response is detailed activity, param sent should
```

Comment:
@jsamoocha i'm much less family with post . The spec says the object being "put" is UpdateableActivity. I am not sure what this would look like in our code - we are sending a dict / json to the API. Where would that object fit in? would we want to use it and serialize vs creating a params dict and sending directly?

ANd is there any benefit in this vs just leaving this all as is given it works and it's dict --> JSON when it is sent to the rest endpoint. 

i hope that makes sense.

Reply:
ok perfect.